samples/subsys/ipc/openamp/sysbuild.cmake

```diff
@@ -0,0 +1,29 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright 2022 NXP
+
+# Select correct remote target for a given board
```

Comment:
I believe going through Kconfig here would be a cleaner approach.
It can couple the remote board name / dependency to the use of a second remote image.

It avoids several `if ... elseif ... elseif ... else` in CMake.
It allows users to extend the Kconfig downstream with extra remote board names without modifying Zephyr tree, for example like this:
```
# Downstream Kconfig.private:
config SB_BOARD_openamp_remote
	string
	default "my_remote_board" if $(BOARD) = "my_main_board"
```

And finally, it allow us to easily refactor this when we get SoC and boards Kconfig cleaned up in a completely generic part (that can be used in both Zephyr and sysbuild) and a Zephyr specific part.

Did you look at this prototype ?
https://github.com/tejlmand/zephyr/commit/efa77ac2e8507fea7bf4be4a67bd0b42af7899a7

especially:
https://github.com/tejlmand/zephyr/blob/efa77ac2e8507fea7bf4be4a67bd0b42af7899a7/samples/subsys/ipc/openamp/Kconfig.sysbuild
https://github.com/tejlmand/zephyr/blob/efa77ac2e8507fea7bf4be4a67bd0b42af7899a7/samples/subsys/ipc/openamp/sysbuild.cmake

Reply:
I've updated the PR to use Kconfig for the remote board name, based off the prototype you mentioned. Thanks for the heads up, I was unaware of `Kconfig.sysbuild`